### DISCUSSION TITLE
Support dynamic routing for pre-generated index page

### DISCUSSION THREAD

**Discussion by a-marcel** [vercel/next.js#21195] [Ideas]
**Describe the feature you'd like to request**

Hello,

i'm not sure if this is a bug or more a feature. I would like to pre-generate some pages from a CMS system, including the index page. At the same time, for all not generated pages i would need dedicated page logic.

**Describe the solution you'd like**

I prepared a example repo: https://github.com/marcelalburg/nextjs-dynamic-index

My expectation is that i see following content:

http://localhost:3005/ -> [static].js -> **Not Working**
http://localhost:3005/home -> [static].js -> Works
http://localhost:3005/test -> [...index].js -> **Not Working**
http://localhost:3005/test/test -> [...index].js -> Works

**Describe alternatives you've considered**

Currently i duplicate the logic (via imports from [static].js) inside an index.js and [static].js page, but this is not that nice.

**Comment by juanferreras** [comment]
Hi! You can use _Optional catch all routes_ for this very purpose.

> Catch all routes can be made optional by including the parameter in double brackets ([[...slug]]).
>
> For example, pages/post/[[...slug]].js will match /post, /post/a, /post/a/b, and so on.
>
> The main difference between catch all and optional catch all routes is that **with optional, the route without the parameter is also matched** (/post in the example above).

https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes

In your case, the exact thing you're looking for is the main difference between _catch all_ `[...index.js]` and _optional catch all_ `[[...index.js]]` (the optional will also work for `/`)

**Reply by a-marcel** [reply]
Hello, thanks for your fast answer. My usecase was, that i've as example 3 pages that are coming from a CMS - including the index page. These pages, i would like to pregenrate via SSG. All other pages will not have anything to do with the CMS system and contains a complete different logic. 

Therefor i would need something like dynamic pre-generated pages that have a fallback to the catch all route.

Thanks a lot

**Comment by colinbull** [comment]
@a-marcel I know this was asked a while ago but one way to solve this is by making `getStaticPaths` emit two slugs for your index entry. 

```
export const getStaticPaths: GetStaticPaths = async () => {
    const paths = await getAllPaths()
    const slugs = paths.flatMap(p => {
        if(p.length === 1 && p[0] === 'index') {
            return [
                {params: {slug: []}},
                {params: {slug: ['index']}}
            ]

        }
        
        return [{params: {slug: p}}]
    })

    return {
        paths: slugs,
        fallback: false
    }
}```
